### PR TITLE
Issue #180 Issue error on circular reference in options file

### LINKED ISSUE #180: RPI 1606: tz390 circular references in options file
Raised by Abe Kornelis 2017/01/29
Circular reference of options file causes unspecified exception in tz390

### PR DESCRIPTION
**Pull request by abekornelis**
#180 Circular reference of options file causes unspecified exception in tz390.

The bat file is incomplete and the equivalent shell file has not been created.

This pull request resolves Issue #180 RPI 1606: tz390 circular references in options file

### CODE REVIEW

File: rt/opt/testopt2L.opt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+* Test with a valid subfile
+  chksrc(2)  * Check sources for illegal characters
+@rt/opt/testopt0.opt
```
Comment (jyganci) [inline rt/opt/testopt2L.opt]: Rename to testopt2L.OPT.
Remove ".opt" on last line.

File: rt/opt/testopt3L.opt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+* Test with an invalid subfile
+  chksrc(2)  * Check sources for illegal characters
+@rt/opt/testoptX.opt
```
Comment (jyganci) [inline rt/opt/testopt3L.opt]: Rename to testopt3L.OPT.
Remove ".opt" on last line.

File: rt/opt/testopt4L.opt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+* Test with a self-reference
+  chksrc(2)  * Check sources for illegal characters
+@rt/opt/testopt4L.opt
```
Comment (jyganci) [inline rt/opt/testopt4L.opt]: Rename to testopt4L.OPT.
Remove ".opt" on last line.

File: rt/opt/testopt5L.opt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+* Test with a circular reference
+  chksrc(2)  * Check sources for illegal characters
+@rt/opt/testopt5LA.opt
```
Comment (jyganci) [inline rt/opt/testopt5L.opt]: Rename to testopt5L.OPT.
Remove ".opt" on last line.

File: rt/opt/testopt5LA.opt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+* Test with a circular reference
+  chksrc(2)  * Check sources for illegal characters
+@rt/opt/testopt5LB.opt
```
Comment (jyganci) [inline rt/opt/testopt5LA.opt]: Rename to testopt5LA.OPT.
Remove ".opt" on last line.

File: rt/opt/testopt5LB.opt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+* Test with a circular reference
+  chksrc(2)  * Check sources for illegal characters
+@rt/opt/testopt5L.opt
```
Comment (jyganci) [inline rt/opt/testopt5LB.opt]: Rename to testopt5LB.OPT.
Remove "opt" on last line.

File: rt/bash/runtestopt [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+opt='@'rt/opt/testopt4L.OPT
+bash/asm rt/mlc/TESTOPT $opt
+
+#
+# this one should assemble with error for circular reference
+echo " "
+echo test 6
+opt='@'rt/opt/testopt5L.OPT
+bash/asm rt/mlc/TESTOPT $opt
+
```
Comment (jyganci) [inline rt/bash/runtestopt]: Script is based on bat version, so has same shortcomings as described below for the bat version.

File: rt/bat/RUNTESTOPT.BAT [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+call bat\asm   rt\mlc\testopt @rt\opt\testopt3.opt
+
+rem
+rem this one should assemble with error for self-reference
+call bat\asm   rt\mlc\testopt @rt\opt\testopt4.opt
+
+rem
+rem this one should assemble with error for circular reference
+call bat\asm   rt\mlc\testopt @rt\opt\testopt5.opt
+
```
Comment (jyganci) [inline rt/bat/RUNTESTOPT.BAT]: While the bat file does run to completion, it requires user interaction to determine whether it runs successfully, so it cannot be added to the BUILD.BAT tests. Anthony created more robust bat files that are invoked in BUILD.BAT. See bat/RUNASSIST.BAT for one example.

Comment (abekornelis) [reply rt/bat/RUNTESTOPT.BAT]: Obviously, the script should test for both the correct assemblies, which is easy, and for the correct error messages on the failing assemblies. The latter is something I do not know how to achieve in a bat file script. I would really appreciate if someone helps me set up the bat file for doing these tests automagically.

File: rt/bat/readme.txt [hunk 9]
```diff
@@ -1,17 +1,18 @@
...
-The currect regressopm test commands are:
+The current regression test commands are:
 RUNASMTESTS - verifies assembler instruction execution
 RUNCBLTESTS - verifies zcobol instructions
 RUNRTQSAM   - verifies QSAM support for large blocks over 32k
 RUNSORT     - verifies sort merge
 RUNZPAR     - verify Zcobol Program Analysis Report
 ZOPCHECK    - verify all z390 assembler instructions generate correct code
 RUNTBRCTX   = verify issue #238 which has been closed.
+RUNTESTOPT  = verify indirection usage in options files
```
Comment (jyganci) [inline rt/bat/readme.txt]: A nit. Change '=' to '-' on the 2 lines.

**Review by jyganci — DISMISSED** [review]
Looks okay.

**Review by dsh33782 — DISMISSED** [review]
Looks ok to me

**Review by dsh33782 — APPROVED** [review]
Looks ok to me, and java changes should merge ok.

**Review by jyganci — APPROVED** [review]
Looks okay.

**Pull request merged**